app-crypt/p11-kit/p11-kit-0.23.22.ebuild

```diff
@@ -0,0 +1,69 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+# Flatcar: inherit systemd eclass for the systemd user unit directory
+# getter, and bash-completion-r1 for bash completion directory getter.
+inherit multilib-minimal systemd bash-completion-r1
+
+DESCRIPTION="Provides a standard configuration setup for installing PKCS#11"
+HOMEPAGE="https://p11-glue.github.io/p11-glue/p11-kit.html"
+SRC_URI="https://github.com/p11-glue/p11-kit/releases/download/${PV}/${P}.tar.xz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~alpha amd64 arm arm64 hppa ~ia64 ~m68k ~mips ppc ppc64 ~riscv ~s390 sparc x86 ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
+IUSE="+asn1 debug +libffi systemd +trust"
+REQUIRED_USE="trust? ( asn1 )"
+
+RDEPEND="asn1? ( >=dev-libs/libtasn1-3.4:=[${MULTILIB_USEDEP}] )
+	libffi? ( dev-libs/libffi:=[${MULTILIB_USEDEP}] )
+	systemd? ( sys-apps/systemd:= )
+	trust? ( app-misc/ca-certificates )"
+DEPEND="${RDEPEND}"
+BDEPEND="virtual/pkgconfig"
+
+pkg_setup() {
+	# disable unsafe tests, bug#502088
+	export FAKED_MODE=1
+}
+
+src_prepare() {
+	if [[ ${CHOST} == *-solaris2.* && ${CHOST##*-solaris2.} -lt 11 ]] ; then
+		# Solaris 10 and before doesn't know about XPG7 (XOPEN_SOURCE=700)
+		# drop to XPG6 to make feature_tests.h happy
+		sed -i -e '/define _XOPEN_SOURCE/s/700/600/' common/compat.c || die
+		# paths.h isn't available, oddly enough also not used albeit included
+		sed -i -e '/#include <paths.h>/d' trust/test-trust.c || die
+		# we don't have SUN_LEN here
+		sed -i -e 's/SUN_LEN \(([^)]\+)\)/strlen (\1->sun_path)/' \
+			p11-kit/server.c || die
+	fi
+	default
+}
+
+multilib_src_configure() {
+	# Flatcar: Override the detection of the systemd user unit
+	# directory and bash completion directory with these
+	# environment variables.
+	local -x systemduserunitdir=$(systemd_get_userunitdir)
+	local -x bashcompdir=$(get_bashcompdir)
```

Comment:
I am wondering if there would be a simpler approach, e.g.

```
econf \
...
--with-systemduserunitdir="$(systemd_get_userunitdir)" \
--with-bashcompdir="$(get_bashcompdir)"
```

Reply:
I rechecked the p11-kit configure script to be sure. There are only options like `--with-bash-completion` and `--with-systemd`, and those simply enable or disable things without taking any values. I can see that `systemduserunitdir` and `bashcompdir` are set up with the [`PKG_CHECK_VAR`](https://autotools.io/pkgconfig/variables.html#idm46482193783040) macro, which adds the following snippet to the `configure --help` output:

```
Some influential environment variables:
  …
  systemduserunitdir
              value of systemduserunitdir for systemd, overriding pkg-config
  bashcompdir value of completionsdir for bash-completion, overriding
              pkg-config
```

So looks like that environment variables are the only way to override the detection. It's less elegant than passing some `--with-foo` options, but I suppose it's still within "that's how we do stuff in autotools idio(ma)tically". :)